core/common/src/main/java/alluxio/underfs/UnderFileSystemFactoryRegistry.java

```diff
@@ -129,13 +145,55 @@ public static UnderFileSystemFactory find(String path) {
         eligibleFactories.add(factory);
       }
     }
+    if (eligibleFactories.isEmpty()) {
```

Comment:
it seems we only scan extensions when eligible factories are empty.
is this expected?